### PR TITLE
Annotate `java.lang.annotation`.

### PR DESCRIPTION
**Pull request by cpovirk**

### CODE REVIEW

File: src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java [hunk 1]
```diff
@@ -45,7 +49,7 @@ public class AnnotationFormatError extends Error {
      *
      * @param   message   the detail message.
      */
-    public AnnotationFormatError(String message) {
+    public AnnotationFormatError(@Nullable String message) {
```
Comment (cpovirk) [inline src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java:52]: This was the one place that jumped out to me as possible controversial: The class documents that `cause` can be null, but it's silent on `message`. I think there's a lot of precedent for the idea that `message` can be null for exceptions in general. But I also doubt that anyone will ever care what we do here, so I'm happy to fold and leave it non-nullable if you think it's better to err on that side in the absence of docs.

**Comment by cpovirk** [comment]
(And as always, I can follow with an eisop PR.)

File: src/java.base/share/classes/java/lang/annotation/IncompleteAnnotationException.java [hunk 2]
```diff
@@ -25,6 +25,9 @@
 
 package java.lang.annotation;
 
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
```
Comment (wmdietl) [inline src/java.base/share/classes/java/lang/annotation/IncompleteAnnotationException.java:29]:
```suggestion
```

Comment (wmdietl) [reply src/java.base/share/classes/java/lang/annotation/IncompleteAnnotationException.java:29]: No use of `@Nullable` was added. Did you mean to use it somewhere?

File: src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java (hunk 1 above)
Comment (wmdietl) [reply src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java:52]: Using `@Nullable` is consistent with what we do elsewhere with `message`s, so I think it's good to add it. We should probably revisit that decision globally, as null is not the best message.

**Review by wmdietl — APPROVED** [review]
Looks good! Do you simply want to add the `@NullMarked` to the `package-info.java` instead of marking every individual file?

File: src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java (hunk 1 above)
Comment (cpovirk) [reply src/java.base/share/classes/java/lang/annotation/AnnotationFormatError.java:52]: Thanks. It feels a little like `WeakReference(@Nullable T t, ...)`: Nothing will blow up if you pass `null`, but it's suspicious that you would want to.

File: src/java.base/share/classes/java/lang/annotation/IncompleteAnnotationException.java (hunk 2 above)
Comment (cpovirk) [reply src/java.base/share/classes/java/lang/annotation/IncompleteAnnotationException.java:29]: Nope, I'd just pasted both imports into every file to start with, and I missed removing it here. Thanks.

**Comment by wmdietl** [comment]
Having tool support to handle additional classes and methods would be good, I agree.

Re-visiting `java.lang` sounds good.

**Pull request merged**